build_resolvers/lib/src/build_asset_uri_resolver.dart

```diff
@@ -102,26 +107,30 @@ class BuildAssetUriResolver extends UriResolver {
       if (resourceProvider.getFile(path).exists) {
         resourceProvider.deleteFile(path);
       }
-      return _AssetState.removed(path);
+      return _AssetState(path, const []);
     }
     globallySeenAssets.add(id);
     transitivelyResolved?.add(id);
     final digest = await buildStep.digest(id);
     if (_cachedAssetDigests[id] == digest) {
-      return _AssetState.unchanged(path, _cachedAssetDependencies[id]);
+      return _AssetState(path, _cachedAssetDependencies[id]);
     } else {
       final isChange = _cachedAssetDigests.containsKey(id);
       final content = await buildStep.readAsString(id);
-      _cachedAssetDigests[id] = digest;
-      final dependencies =
-          _cachedAssetDependencies[id] = _parseDirectives(content, id);
+      // ignore: invariant_booleans
+      if (_cachedAssetDigests[id] == digest) {
```

Comment:
I think the lint was actually correctly triggering here? This is effectively unreachable code (will always evaluate to false?)

Reply:
NVM I see now that this is to check if the asset digest was updated after the `await buildStep.readAsString`.

I would add a comment about that to clarify

Reply:
I don't think it was correct. I need to file a bug against the lint. The `await` means that someone else could have updated `_cachedAssetDigests` since we checked it last.

Reply:
This is somewhat similar to https://github.com/dart-lang/linter/issues/1858